libs/x-content/src/test/java/org/elasticsearch/xcontent/XContentParserTests.java

```diff
@@ -655,6 +657,25 @@ public void testCreateRootSubParser() throws IOException {
 
     }
 
+    public void testJsonIncludeSourceOnParserError() throws IOException {
+        var xContent = XContentFactory.xContent(XContentType.JSON);
+        var source = "{\"field\": invalid}"; // causes parse exception
+        var sourceEnabled = XContentParserConfiguration.EMPTY;
+        var sourceDisabled = XContentParserConfiguration.EMPTY.withIncludeSourceOnError(false);
+
+        var parseException = assertThrows(XContentParseException.class, () -> createParser(xContent, sourceEnabled, source).map());
```

Comment:
nit: assertThrows -> expectThrows (the former is from junit, and doesn't produce as clear error messages)